src/python/views/demo_tigergraph_circle/__init__.py

```diff
@@ -0,0 +1,774 @@
+
+
+# Given URL params, render left sidebar form and return combined filter settings
+# https://docs.streamlit.io/en/stable/api.html#display-interactive-widgets
+def sidebar_area():
+    # q = conn.runInstalledQuery("mostDirectInfections")
+    # Most_infectious_IDS = q[0]['Answer']
+    # MI_List = [d['v_id'] for d in Most_infectious_IDS if 'v_id' in d]
+    num_edges_init = urlParams.get_field('num_matches', 0.5)
+    # MI_List.reverse()
+    idList = [i for i in range(1, 500)]
+    st.sidebar.header("TigerGraph Anti-Fraud")
+    tg_host = st.sidebar.text_input ('TigerGraph Host')
+    tg_username = st.sidebar.text_input ('TigerGraph Username')
+    tg_password = st.sidebar.text_input ('TigerGraph Password')
+    tg_graphname = st.sidebar.text_input ('TigerGraph Graphname')
+    tg_secret = st.sidebar.text_input('TigerGraph Secret')
+    
+    if st.sidebar.button("Connect"):
+        try:
+            conn = tg.TigerGraphConnection(host=tg_host, graphname=tg_graphname, username=tg_username, password=tg_password)
+            if tg_secret:
+                conn.getToken(tg_secret)
+            else:
+                conn.getToken(conn.createSecret())
+            st.sidebar.success("Connnected Successfully")
+            user_id = st.sidebar.selectbox(
+                'User ID ',
+                idList
+            )
+            urlParams.set_field('user_id', user_id)
+
+            return {'user_id': user_id, 'conn': conn}
+
+        except:
+            st.sidebar.error("Failed to Connect")
+            return None
+    
+    return None
+
+    user_id = st.sidebar.selectbox(
+        'User ID ',
+        idList
+    )
+
+    urlParams.set_field('user_id', user_id)
+
+    return {'user_id': user_id, 'conn': conn}
+
+
+def plot_url(nodes_df, edges_df):
+    global metrics
+
+    logger.info('Starting graphistry plot')
+    tic = time.perf_counter()
+
+    # edge weight ( ==> score )
+    # edgeInfluence @ https://hub.graphistry.com/docs/api/1/rest/url/#urloptions
+
+    g = graphistry \
+        .edges(edges_df) \
+        .settings(url_params={'play': 7000, 'dissuadeHubs': True}) \
+        .bind(edge_weight='amount')      \
+        .bind(source='from_id', destination='to_id') \
+        .bind(edge_title='amount', edge_label='amount') \
+        .nodes(nodes_df) \
+        .bind(node='n') \
+        .addStyle(bg={'color': 'white'}) \
+        .encode_point_color("color") \
+        .encode_edge_color("color") \
+        .encode_point_icon('type', categorical_mapping={'User': 'laptop',
+                                                        'Transaction': 'server'},
+                           default_mapping='question')
+
+
+# .encode_point_size('', ["blue", "yellow", "red"],  ,as_continuous=True)
+    # if not (node_label_col is None):
+    #     g = g.bind(point_title=node_label_col)
+
+    # if not (edge_label_col is None):
+    #     g = g.bind(edge_title=edge_label_col)
+
+    url = g.plot(render=False)
```

Comment:
next rev of g-a-k will be pygraphistry 0.15, where we can update this to `g.plot(render=False, as_files=True)` to make it go even faster..

Reply:
Updated in latest push